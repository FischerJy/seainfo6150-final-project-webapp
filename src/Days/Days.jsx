import React, { createRef, useEffect, useState } from "react";
import { useHistory } from "react-router-dom";

export default function Days() {
  const [name, setName] = useState("");

  const [info, setInfo] = useState({
    coord: {
      lat: "",
      lon: "",
    },
  });
  const [state, setState] = useState("");

  const node = createRef();
  const his = useHistory();

  useEffect(() => {
    var getlist = (name) => {
      fetch(
        `https://api.openweathermap.org/data/2.5/weather?q=${name}&appid=d2c0715f020c3a6cd3e893289a84cbc7&units=metric`
      )
        .then((res) => res.json())
        .then((res) => {
          console.log("res -> :", res);
          if (res.cod === "404") {
            setState(res.message);
          } else {
            his.push(`/days/${name}/${res.coord.lat}/${res.coord.lon}`);
          }
        });
    };
    name && getlist(name);
    return () => {
      getlist = () => {};
    };
  }, [his, name]);

  return (
    <div className="main">
      <br />
      <br />
      {state ? (
        <div style={{ textAlign: "center" }}>{state}</div>
      ) : (
        <p style={{ textAlign: "center" }}>
          name: <input type="text" ref={node} />
          <button
            onClick={() => {
              setName(node.current.value);
            }}
          >
            Search
          </button>
        </p>
      )}
    </div>
  );
}
