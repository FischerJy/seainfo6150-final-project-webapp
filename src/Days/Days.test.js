import React from "react";
import { render } from "@testing-library/react";
import Days from "./Days.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the Days component correctly", () => {
    const { container } = render(<BrowserRouter>
        <Days />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});