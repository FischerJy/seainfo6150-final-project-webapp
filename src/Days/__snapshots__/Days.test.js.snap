// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Days component correctly 1`] = `
<div>
  <div
    class="main"
  >
    <br />
    <br />
    <p
      style="text-align: center;"
    >
      name: 
      <input
        type="text"
      />
      <button>
        Search
      </button>
    </p>
  </div>
</div>
`;
